pkg/kudoctl/kudoinit/types.go

```diff
@@ -25,10 +25,7 @@ type Artifacter interface {
 
 type InstallVerifier interface {
 	// PreInstallVerify verifies that the installation is possible
-	PreInstallVerify(client *kube.Client) verifier.Result
-
-	// TODO: Add verification of existing installation
-	// VerifyInstallation(client *kube.Client) Result
+	PreInstallVerify(client *kube.Client, result *verifier.Result) error
```

Comment:
How about returning `(verifier.Result, error)` instead? This way there would be a clear distinction between input and output values. Also, currently most of the times when `PreInstallVerify` is called we have something like this:

```go
result := verifier.NewResult()
err = crd.NewInitializer().VerifyInstallation(kubeClient, &result)
```

whereas

```go
result, err := crd.NewInitializer().VerifyInstallation(kubeClient)
```

is much clearer IMO. For the code in `crds.go` to merges multiple `Result`s, we could then use `result.Merge`. WDYT?

Reply:
Yeah... I'm not sure, to be honest. This was basically the pattern I had before, and I do agree it looks nicer on the caller side.

Passing the result in allow the handling inside the funcs to look like this though:

```
	if err := o.validateServiceAccountExists(client, result); err != nil {
		return err
	}
```

If were to return (verifier.Result, error) it would have to look like this:
```
    myRes := verifier.NewResult{}
    res, err := o.validateServiceAccountExists(client)
    if err != nil {
        return err
    }
    myRes.merge(res)
   
    // ... next check, validate SA permissions
    return res
```

So... It might be an option to change the signature for the Top-Level methods, but that is at the cost of consistency.  

If other devs feels strongly about this as well I'll change it, but at the moment I prefer it like this.

Reply:
Yeah, I see how having multiple return values complicates things a bit, we can't do

```go
result.Merge(o.validateServiceAccountExists(client))
```

I don't feel strongly about this, happy to keep it as it is right now.